src/Library/EventPublisher.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Microsoft.IstioMixerPlugin.Library
+{
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.IstioMixerPlugin.Common;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+
+    internal class EventPublisher
+    {
+        private IHeartbeatPropertyManager heartbeatModule;
+        private bool firstTime = true;
+        public EventPublisher()
+        {
+            var telemetryModules = TelemetryModules.Instance;
+            this.heartbeatModule = telemetryModules.Modules.OfType<IHeartbeatPropertyManager>().FirstOrDefault();
+            Diagnostics.LogInfo("EventPubliesher initialized");
+        }
+
+        public bool UpdateClusterId(string clusterId)
```

Comment:
Can you please add a note that this is not thread-safe? We might be calling it from some sort of handler, so it might end up being relevant.